src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/AccountSupervisor.kt

```diff
@@ -767,20 +769,29 @@ internal open class AccountSupervisor(
             val result = helper.parser.decodeJsonObject(additionalPayload)
 
             if (error == null && result != null) {
-                val url = complianceGeoblockUrl()
                 val signedMessage = helper.parser.asString(result["signedMessage"])
                 val publicKey = helper.parser.asString(result["publicKey"])
                 val timestamp = helper.parser.asString(result["timestamp"])
+                val isKeplr = helper.parser.asBool(result["isKeplr"])
+                val url = if (isKeplr == true) complianceGeoblockKeplrUrl() else complianceGeoblockUrl()
 
                 val isUrlAndKeysPresent =
                     url != null &&
                         signedMessage != null &&
-                        publicKey != null &&
-                        timestamp != null
+                        publicKey != null && (timestamp != null || isKeplr == true)
```

Comment:
[fix/suggestion] would prefer to separate the `timestamp!=null || isKelper == true` into a separately named variable, or rename `isUrlAndKeysPresent` to explain more of what's happening here.